consensus/common_test.go

```diff
@@ -195,19 +195,8 @@ func decideProposal(
 	height int64,
 	round int32,
 ) (proposal *types.Proposal, block *types.Block) {
-	oldPrivValidator := cs1.privValidator
-	oldPrivValidatorPubKey := cs1.privValidatorPubKey
 	cs1.mtx.Lock()
-	pubKey1, _ := cs1.privValidator.GetPubKey()
-	pubKey2, _ := vs.PrivValidator.GetPubKey()
-	if !pubKey1.Equals(pubKey2) {
-		// block creator must be the cs.privValidator
-		cs1.privValidator = vs.PrivValidator
-		cs1.privValidatorPubKey = pubKey2
```

Comment:
These technics are happened `deadlock` by `cs1.createProposalBlock(round)` with `test --race`. And I made `createProposalBlockSlim`.